drivers/input/input_gt911.c

```diff
@@ -289,11 +316,12 @@ static int gt911_init(const struct device *dev)
 
 #define GT911_INIT(index)                                                     \
 	static const struct gt911_config gt911_config_##index = {	       \
-		.bus = I2C_DT_SPEC_INST_GET(index),			       \
 		.rst_gpio = GPIO_DT_SPEC_INST_GET(index, reset_gpios), \
 		.int_gpio = GPIO_DT_SPEC_INST_GET(index, irq_gpios)	       \
 	};								       \
-	static struct gt911_data gt911_data_##index;			       \
+	static struct gt911_data gt911_data_##index = {			       \
+		.bus = I2C_DT_SPEC_INST_GET(index),			       \
+	};								       \
```

Comment:
I think that partially initializing the structure here causes the linker to hold a whole copy of it in flash, which is a bit wasteful. At this point you may as well statically initialize everything else in that structure but that's also not very elegant in my opinion.

How about:
- write the alternate address in the devicetree and save it `gt911_config`
- add an "actual address" in `gt911_data`, populate it in `gt911_init` by probing first `config->bus.addr` and then `config->alt_addr` (which you already do here)
- use a couple of driver specific read/write wrapper instead of the standard `_dt` ones, that use the bus from `config->bus` but the address from `data->actual_address`

Reply:
Sure, I'm fine with this approach. I'll update the PR to use this method.

Reply:
Thanks, another thought after your comment above is that you could make the alternate address DT property optional, so one may not set it at all if the specific design has a well define strapping, and then you skip the whole alternate address thing based on that.

Reply:
Yup, that's what I did- if the `alt_addr` property is zero then the alternate address code is skipped